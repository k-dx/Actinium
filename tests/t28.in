let i = -10;
let f = fn () {
    fn (x) {
        let i = 0;
        fn (y) {
            i = i + 3;
            print i;
            print x + y;
            x + y;
        };
    };
};
let c = f();
c(1)(2); // => 1 3
print "---";
c(4)(2); // => 1 6
print "---";
print i; // => -10
print "---";
let d = f()(10);
d(1); // => 3 11
print "---";
d(2); // => 6 12