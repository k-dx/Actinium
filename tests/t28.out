3
3
---
3
6
---
-10
---
3
11
---
6
12